src/EventSubscriber/RequestSubscriber.php

```diff
@@ -263,4 +274,35 @@ protected function constructQuerySpan($connection, array $query) {
     return $span;
   }
 
+  /**
+   * Add options to pass to the PHP Agents.
+   *
+   * @return array
+   *   An array of options to pass to the PHP Agent. Ie. tags.
+   */
+  protected function getRequestOptions() {
+    $options = [];
+
+    // Fetch the current path.
+    $route_object = $this->routeMatch->getRouteObject();
+    $path = $route_object->getPath();
+    $route_options = $route_object->getOptions();
+
+    // If this is an admin page, add a custom variable to denote that.
+    if (isset($route_options['_admin_route'])) {
+      $options['tags']['admin_page'] = TRUE;
+    }
+
+    // Add tags depending on the page we're in.
+    foreach (ELASTIC_APM_PARENT_ROUTES as $route) {
+      if (strpos($path, $route) === FALSE) {
+        continue;
+      }
```

Comment:
**CHANGE REQUEST**

We will need to make this configurable. I have created separate issues for this.

Have you tested this code? The path is not a very reliable way because, for example, the path to the cart page is `/cart` and it won't match. The best is to use the route names and we'll need tag/route-patterns pairs i.e.

```
[
  'commerce' => [
    'entity.commerce_product.*',
    'commerce_cart.*',
    'commerce_checkout.*',
  ],
  'search' => [
    'entity.search_page.*',
  ],
];
```

That will also play well with the changes when we make it configurable, see related D.org issues.